internal/providers/msentraid/msentraid.go

```diff
@@ -111,6 +117,15 @@ func (p Provider) userClaims(idToken *oidc.IDToken) (claims, error) {
 
 // getGroups access the Microsoft Graph API to get the groups the user is a member of.
 func (p Provider) getGroups(token *oauth2.Token) ([]info.Group, error) {
+	// Check if the token has the GroupMember.Read.All scope
+	scopes, err := p.getTokenScopes(token)
+	if err != nil {
+		return nil, err
+	}
+	if !slices.Contains(scopes, "GroupMember.Read.All") {
+		return nil, errors.New("app is missing the GroupMember.Read.All permission")
```

Comment:
IMO, `app` does not say much here. Can we be a bit clearer regarding what's missing the permission?

Reply:
How about `The Microsoft Entra ID app is missing the GroupMember.Read.All permission`?

Reply:
Also, if we want to display this error to the user instead of just logging it, we need to implement special case handling for it.

Reply:
> Also, if we want to display this error to the user instead of just logging it, we need to implement special case handling for it.

Done in https://github.com/ubuntu/authd-oidc-brokers/pull/135/commits/0be47303180b16b239c805cdd0376179bd00d0ad